src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java

```diff
@@ -0,0 +1,233 @@
+
+        response.setCredentialIssuer("");
+        response.setCredentialEndPoint("http://example.com");
+
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertEquals("RESIDENT-APP-041", invalidWellknownResponseException.getErrorCode());
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("credentialIssuer: must not be blank"), "Exception message should indicate the missing 'credentialIssuer'");
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenCredentialEndpointIsIncorrectInCredentialIssuerWellknownResponse() throws ApiNotAccessibleException {
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+
+        response.setCredentialEndPoint("http://example.com"); // Incorrect because it does not end with "/credential"
+
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertEquals("RESIDENT-APP-041", invalidWellknownResponseException.getErrorCode());
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("credentialEndPoint: must match \"https?://.*?/credential$\""), "Exception message should indicate the incorrect 'credentialEndpoint'");
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenAuthorizationServersIsEmpty() {
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+
+        response.setCredentialIssuer("https://valid.url/credential");
+        response.setAuthorizationServers(Collections.emptyList());  // Invalid empty list
+
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertEquals("RESIDENT-APP-041", invalidWellknownResponseException.getErrorCode());
```

Comment:
Can we import Assertions and access the assertEquals directly

Reply:
yes, @abhip2565 let's import Assertions

Reply:
done